### PR TITLE
Feature: CRUD study entity

### PR DESCRIPTION
**Pull request by chinyakao**
- Create study
    - enable notification -> create aws topic for notification
    - disable notification -> create study as a folder of survey list
    - collect activity data -> select the time period to collect
- Read study
    - get all the studies belong to the accounts
- Update study title
- Delete study
    - enable notification -> delete aws topic's subscription and aws topic, then delete study
    - disable notification -> directly delete study

### CODE REVIEW

**Comment by chinyakao** [comment]
@soumyaray @ShellyChaoo 
Hi everyone, here is the quick update about `study` entity. 
I am not sure this sturcture(architecture), especially the new files in `services\study` folder, in this PR. 
Sometimes, it will link to datamapper to do some work. Sometimes, it will directly link to database_repositories to do some work. If there is something wrong, please let me know.
Also, the `study_mapper.rb` is not in the `google_mappers` folder. Because it might not directly related to google, it is more directly related to database. But, if it is not correct, please let me know.
Thank you so much!

File: application/controllers/app.rb [hunk 1]
```diff
@@ -469,6 +469,114 @@ class App < Roda
...
+        # GET survey/[survey_id]/close
+        # routing.get 'close' do
+        #   response = Service::CloseSurvey.new.call(survey_id: survey_id)
+
+        #   flash[:error] = response.failure + ' Please try again.' if response.failure?
+
+        #   routing.redirect '/survey_list'
+        # end
+
+        # DELETE survey/[survey_id]
```
Comment (soumyaray) [inline application/controllers/app.rb:562]: should it be `study/[study_id]`?

File: application/services/studies/create_study.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        input[:study] = study
+
+        Success(input)
+      rescue
+        Failure('Failed to store into database.')
+      end
+
+      def get_study_arn(input)
+        if input[:params]['enable_notification']
+          study_arn = Messaging::Notification.new(input[:config]).create_topic(input[:study][:id])
```
Comment (chinyakao) [inline application/services/studies/create_study.rb:37]: Directly talk to messaging object to create the aws topic and get the aws arn

File: application/services/studies/delete_study.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        Success(input)
+      rescue StandardError => e
+        puts e
+        Failure('Failed to get study arn from database.')
+      end
+
+      # input { config:, study_id:, aws_arn: }
+      def delete_aws_topic(input)
+        study = input[:study]
+        Messaging::Notification.new(input[:config]).delete_topic(study.aws_arn) if study.enable_notification
```
Comment (chinyakao) [inline application/services/studies/delete_study.rb:33]: Directly talk to messaging object to delete the aws topic

File: application/services/studies/create_study.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+      private
+
+      def store_into_database(input)
+        track_activity = !input[:params]['track_activity'].nil?
+        enable_notification = !input[:params]['enable_notification'].nil?
+        input[:params].update('track_activity' => track_activity,
+                              'enable_notification' => enable_notification,
+                              'aws_arn' => 'checking enable notification or not')
+        new_study = Mapper::StudyMapper.new.load(input[:params], input[:current_account])
```
Comment (chinyakao) [inline application/services/studies/create_study.rb:26]: Talk to mapper obj only when creating a study. Other cases(read, delete and update) did not talk to mapper obj.
Not sure if the timing is correct to use mapper like this.

File: application/controllers/app.rb [hunk 5]
```diff
@@ -469,6 +469,53 @@ class App < Roda
...
+
+            routing.redirect '/studies', 303
+          end
+        end
+
+        # GET /studies
+        routing.get do
+          routing.redirect '/' unless @current_account
+
+          studies = Repository::For[Entity::Study].find_owner(@current_account['id'])
```
Comment (chinyakao) [inline application/controllers/app.rb:514]: Here is no service obj but just directly talking to repository.

**Comment by ShellyChaoo** [comment]
some changes on develop branch

**Review by chinyakao — COMMENTED** [review]
Four questions in the comments

**Pull request merged**